packages/@react-spectrum/tabs/docs/Tabs.mdx

```diff
@@ -112,6 +109,94 @@ function Example() {
 To internationalize Tabs, a localized string should be passed as children to the TabList `<Item>`. Any text content within the Tab's panel should also be localized accordingly.
 For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of Tabs is automatically flipped.
 
+### Icons in tabs
+
+Icons can also be used in `Tabs` in addition to a text label. Tabs use [Slots](./layout.html#slots) in order to style Icons and Text together.
+
+```tsx example
+<Tabs aria-label="History of Ancient Rome">
+  <TabList>
+    <Item key="FoR"><Bookmark /><Text>Founding of Rome</Text></Item>
+    <Item key="MaR"><Calendar /><Text>Monarchy and Republic</Text></Item>
+    <Item key="Emp"><Dashboard /><Text>Empire</Text></Item>
+  </TabList>
+  <TabPanels>
+    <Item key="FoR">
+      Arma virumque cano, Troiae qui primus ab oris.
+    </Item>
+    <Item key="MaR">
+      Senatus Populusque Romanus.
+    </Item>
+    <Item key="Emp">
+      Alea jacta est.
+    </Item>
+  </TabPanels>
+</Tabs>
+```
+
+### Customizing layout
+
+If you need to add additional structure to `Tabs` such as buttons in-line with the `TabList`, we support this too.
+`TabList` and `TabPanels` are not required to be the immediate children of `Tabs`.
+
+```tsx example
+function Example () {
+  let [tabs, setTabs] = React.useState([
+    {name: 'Tab 1', children: 'Tab Body 1'},
+    {name: 'Tab 2', children: 'Tab Body 2'},
+    {name: 'Tab 3', children: 'Tab Body 3'}
+  ]);
+  let addTab = () => {
+    let newTabs = [...tabs];
+    newTabs.push({
+      name: `Tab ${tabs.length + 1}`,
+      children: `Tab Body ${tabs.length + 1}`
+    });
+
+    setTabs(newTabs);
+  };
+
+  let removeTab = () => {
+    if (tabs.length > 1) {
+      let newTabs = [...tabs];
+      newTabs.pop();
+      setTabs(newTabs);
+    }
+  };
+
+  return (
+    <Tabs {...props} width="100%" aria-label="Tab example" items={tabs}>
```

Comment:
the focus ring is a bit cut off here: 
![image](https://user-images.githubusercontent.com/9661127/116761039-6dc4c780-a9cb-11eb-86c7-8e5fda0adc65.png)
maybe need some padding left on the TabList and TabPanels?

Reply:
whoa, i didn't see that locally, but i see it in the built one here
how'd you get it to focus both remove and tab 1?

Reply:
oh, wut didn't even notice that when making the screenshot lol. I can just tab onto the tab and it focuses both of them, hmm

Reply:
reproduced, happens once you delete all the tabs via "Remove Tab"

Reply:
Probably due to the unsafe style below? We really should avoid those... Can we simplify this example?

Reply:
Reason was due to overflow:hidden, i was able to remove that property and use minWidth to keep the flex contained
I simplified it a bit, in order to get rid of the UNSAFE style, I had to turn that node into a div so that I could use border bottom, this is necessary so that the border extends all the way beneath the add/remove buttons